editor/editor_log.cpp

```diff
@@ -379,6 +380,13 @@ void EditorLog::_reset_message_counts() {
 	}
 }
 
+void EditorLog::_check_need_rebuild() {
+	if (need_rebuld && !log->is_updating()) {
+		need_rebuld = false;
+		_rebuild_log();
```

Comment:
Is there anything other than rebuild the whole log ? Performance maybe not so critical here but it's still better to consider it.  Maybe we don't need to rebuild the whole log, just append those get rejected by updating ? 

Ah, the code does look suspicious, how can we ensure we don't encounter the updating in `_rebuild_log` ? Internally it calls `_add_log_line` too.

Reply:
When rebuilding the log. `log->isupdating` will return false. I can also add messages to the cache. Or rebuild directly when `log->isupdating`.

Reply:
Personally I think we can find a way to do some stress testing ？Log a tons of message and make sure eventually they all show in the log console and in the right order.